controllers.js

```diff
@@ -21,46 +21,58 @@ exports.fetchTopics = (req, res, next) => {
 exports.fetchArticle = (req, res, next) => {
   const { article_id } = req.params;
 
-  
   if (isNaN(article_id)) {
-      return res.status(400).send({ message: 'Bad Request!' });
+    return res.status(400).send({ message: 'Bad Request!' })
   }
 
   getArticleById(article_id)
-      .then(article => {
-          if (!article) {
-              return res.status(404).send({ message: 'Article not found' });
-          }
-          res.status(200).send(article);
-      })
-      .catch(err => {
-          console.error(err);
-          next(err);
-      });
+    .then(article => {
+      if (!article) {
+        return res.status(404).send({ message: 'Article not found' })
+      }
+      res.status(200).send(article)
+    })
+    .catch(err => {
+      console.error(err)
+      next(err)
+    })
 };
 
 exports.fetchArticles = (req, res, next) => {
   const { sort_by } = req.query;
 
-  
   const validSortColumns = ['author', 'title', 'article_id', 'topic', 'created_at', 'votes', 'article_img_url', 'comment_count'];
 
-  
   if (sort_by && !validSortColumns.includes(sort_by)) {
     return res.status(400).send({ message: 'Bad Request!' })
   }
 
-  
   const sortColumn = sort_by || 'created_at'
-  
+
   getArticles(sortColumn)
     .then((articles) => {
       res.status(200).send({ articles })
     })
     .catch(next)
+};
+
+exports.fetchComments = (req, res, next) => {
+  const { article_id } = req.params;
+
+  if (isNaN(article_id)) {
+    return res.status(400).send({ message: 'Bad Request!' })
   }
 
+  getCommentsByArticleId(article_id)
```

Comment:
you need to check if an article exists before fetching any comments as an empty array could be for an article that simply has no comments.